From 3b72b0a96e2e2d823410490a007e3dc7e5e06768 Mon Sep 17 00:00:00 2001
From: hongjiang zheng <hongjiang.a.zheng@intel.com>
Date: Sun, 14 Sep 2014 08:45:23 +0800
Subject: [PATCH 148/164] atomisp2: add isp config_id into 3a stats

BZ: 223442

This patch adds the isp_config_id(which is from ISP FW to indicate it is from
which parameter) to the 3A statistics.

Change-Id: Id107ac25d66875e4e6e81dc8f7c8e94c1763d735
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-8929
Signed-off-by: hongjiang zheng <hongjiang.a.zheng@intel.com>
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://android.intel.com:443/260220
Reviewed-on: https://icggerrit.ir.intel.com/29095
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../pci/atomisp2/atomisp_driver/atomisp_cmd.c      |    6 ++++--
 .../atomisp_driver/atomisp_compat_ioctl32.c        |    6 ++++--
 .../atomisp_driver/atomisp_compat_ioctl32.h        |    1 +
 .../pci/atomisp2/atomisp_driver/atomisp_ioctl.c    |    5 +++--
 4 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index da000a2..784f158e 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -2509,6 +2509,8 @@ int atomisp_3a_stat(struct atomisp_sub_device *asd, int flag,
 			s3a_buf->s3a_data);
 
 	config->exp_id = s3a_buf->s3a_data->exp_id;
+	config->isp_config_id = s3a_buf->s3a_data->isp_config_id;
+
 	ret = copy_to_user(config->data, asd->params.s3a_user_stat->data,
 			   asd->params.s3a_output_bytes);
 	if (ret) {
@@ -2520,8 +2522,8 @@ int atomisp_3a_stat(struct atomisp_sub_device *asd, int flag,
 	/* Move to free buffer list */
 	list_del_init(&s3a_buf->list);
 	list_add_tail(&s3a_buf->list, &asd->s3a_stats);
-	dev_dbg(isp->dev, "%s: finish getting exp_id %d 3a stat\n", __func__,
-		config->exp_id);
+	dev_dbg(isp->dev, "%s: finish getting exp_id %d 3a stat, isp_config_id %d\n", __func__,
+		config->exp_id, config->isp_config_id);
 	return 0;
 }
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.c
index 56f9a02..db55b4b 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.c
@@ -250,7 +250,8 @@ static int get_atomisp_3a_statistics32(struct atomisp_3a_statistics *kp,
 		copy_from_user(kp, up, sizeof(struct atomisp_grid_info)) ||
 		get_user(rgby_data, &up->rgby_data) ||
 		get_user(data, &up->data) ||
-		get_user(kp->exp_id, &up->exp_id))
+		get_user(kp->exp_id, &up->exp_id) ||
+		get_user(kp->isp_config_id, &up->isp_config_id))
 			return -EFAULT;
 
 	kp->data = compat_ptr(data);
@@ -270,7 +271,8 @@ static int put_atomisp_3a_statistics32(struct atomisp_3a_statistics *kp,
 		copy_to_user(up, kp, sizeof(struct atomisp_grid_info)) ||
 		put_user(rgby_data, &up->rgby_data) ||
 		put_user(data, &up->data) ||
-		put_user(kp->exp_id, &up->exp_id))
+		put_user(kp->exp_id, &up->exp_id) ||
+		put_user(kp->isp_config_id, &up->isp_config_id))
 			return -EFAULT;
 
 	return 0;
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h
index de31285..f9d3cd4 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h
@@ -67,6 +67,7 @@ struct atomisp_3a_statistics32 {
 	compat_uptr_t data;
 	compat_uptr_t rgby_data;
 	uint32_t exp_id;
+	uint32_t isp_config_id;
 };
 
 struct atomisp_metadata_with_type32 {
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index 31739d0..098f0d0 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -1465,8 +1465,9 @@ static int atomisp_dqbuf(struct file *file, void *fh, struct v4l2_buffer *buf)
 	buf->reserved2 = pipe->frame_config_id[buf->index];
 	rt_mutex_unlock(&isp->mutex);
 
-	dev_dbg(isp->dev, "dqbuf buffer %d (%s) with exp_id %d\n",
-		buf->index, vdev->name, __get_frame_exp_id(pipe, buf));
+	dev_dbg(isp->dev, "dqbuf buffer %d (%s) with exp_id %d, isp_config_id %d\n",
+		buf->index, vdev->name, __get_frame_exp_id(pipe, buf),
+		buf->reserved2);
 	return 0;
 }
 
-- 
1.7.9.5

